lib/tpm2_eventlog_yaml.c

```diff
@@ -250,101 +251,127 @@ static bool yaml_uefi_var(UEFI_VARIABLE_DATA *data, size_t size, UINT32 type) {
     }
 
-    /* PK, KEK, db, and dbx are reserved variables names used to store platform
-     * keys, key exchange keys, database keys, and blacklisted database keys,
-     * respectively. 
-     */
-    if (type == EV_EFI_VARIABLE_DRIVER_CONFIG && 
-        ((strlen(ret) == 2 && strncmp(ret, "PK", 2) == 0) ||
-        (strlen(ret) == 3 && strncmp(ret, "KEK", 3) == 0) ||
-        (strlen(ret) == 2 && strncmp(ret, "db", 2) == 0) ||
-        (strlen(ret) == 3 && strncmp(ret, "dbx", 3) == 0))) {
-
-        free(ret);
-        tpm2_tool_output("      VariableData:\n");
-        uint8_t *variable_data = (uint8_t *)&data->UnicodeName[
-            data->UnicodeNameLength];
-        /* iterate through each EFI_SIGNATURE_LIST */
-        while (start < size) {
-            EFI_SIGNATURE_LIST *slist = (EFI_SIGNATURE_LIST *)variable_data;
-            if (start + sizeof(*slist) > size) {
-                LOG_ERR("EventSize is inconsistent with actual data\n");
-                break;
-            }
-
-            if (slist->SignatureSize < 16) {
-                LOG_ERR("SignatureSize is too small\n");
-                break;
-            }
-
-            uuid_unparse_lower(slist->SignatureType, uuidstr);
-            tpm2_tool_output("      - SignatureType: %s\n"
-                             "        SignatureListSize: %" PRIu32 "\n"
-                             "        SignatureHeaderSize: %" PRIu32 "\n"
-                             "        SignatureSize: %" PRIu32 "\n"
-                             "        Keys:\n",
-                             uuidstr, slist->SignatureListSize,
-                             slist->SignatureHeaderSize,
-                             slist->SignatureSize);
-
-            start += (sizeof(*slist) + slist->SignatureHeaderSize);
-            if (start + slist->SignatureSize > size) {
-                LOG_ERR("EventSize is inconsistent with actual data\n");
-                break;
-            }
-
-            int signature_size = slist->SignatureListSize -
-                sizeof(*slist) - slist->SignatureHeaderSize;
-            if (signature_size < 0 || signature_size % slist->SignatureSize != 0) {
-                LOG_ERR("Malformed EFI_SIGNATURE_LIST\n");
-                break;
+    if (yaml_version == 2) {
+        /* PK, KEK, db, and dbx are reserved variables names used to store platform
+         * keys, key exchange keys, database keys, and blacklisted database keys,
+         * respectively.
+         */
+        if (type == EV_EFI_VARIABLE_DRIVER_CONFIG) {
+            if ((strlen(ret) == 2 && strncmp(ret, "PK", 2) == 0) ||
+                (strlen(ret) == 3 && strncmp(ret, "KEK", 3) == 0) ||
+                (strlen(ret) == 2 && strncmp(ret, "db", 2) == 0) ||
+                (strlen(ret) == 3 && strncmp(ret, "dbx", 3) == 0)) {
+
+                free(ret);
+                tpm2_tool_output("      VariableData:\n");
+                uint8_t *variable_data = (uint8_t *)&data->UnicodeName[
```

Comment:
Is the typecast required?

Reply:
The field in the struct is defined as 
```  
char16_t UnicodeName[];
```
I believe it gave a compiler warning without the explicit typecast.